File: Frontend/ReadNest/src/types/book.ts
export interface Book {
    bookId: number;
    title: string;
    author: string;
    Tags: string[];
    description: string;
    quantity: number;
    cover: string;
}

// the root state
export interface BookRoot {
    books: Book[]
}

//when add a new book
export interface bookData {
    title: string;
    author: string;
    description: string;
    quantity: number | "";
}

//when editing book
export interface EditbookType {
    bookId: number;
    title: string;
    author: string;
    description: string;
    quantity: number | "";
}

File: backend/src/main/java/com/readnest/pojo/Book.java
package com.readnest.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book {
    private Integer bookId;
    private String title;
    private String author;
    private String description;
    private Integer quantity;
    private String cover;
}
